src/Persistence.Tests/Yaml/DeserializerValidTests.cs

```diff
@@ -267,4 +270,29 @@ public void Deserialize_Strings()
     }
 
 
+    [TestMethod]
+    [DataRow(@"_TestData/ValidYaml/Screen/with-gallery.fx.yaml")]
+    public void Deserialize_Should_AddGalleryTemplate(string path)
+    {
+        // Arrange
+        var deserializer = ServiceProvider.GetRequiredService<IYamlSerializationFactory>().CreateDeserializer();
+        using var yamlStream = File.OpenRead(path);
+        using var yamlReader = new StreamReader(yamlStream);
+
+        // Act
+        var screen = deserializer.Deserialize<Screen>(yamlReader);
+
+        // Assert
+        screen.Should().NotBeNull();
+        if (screen.Children == null)
+            throw new ArgumentNullException(nameof(screen.Children));
+        screen.Children.Should().NotBeNull().And.HaveCount(1);
+        var gallery = screen.Children[0];
+        gallery.Should().NotBeNull().And.BeOfType<BuiltInControl>();
+        gallery.Template.Name.Should().Be("Gallery");
+        if (gallery.Children == null)
+            throw new ArgumentNullException(nameof(gallery.Children));
+        gallery.Children.Should().NotBeNull().And.HaveCount(2);
```

Comment:
The if -> throw is redundant with the `.NotBeNull()` FluentAssertion right below it
```suggestion
        gallery.Children.Should().NotBeNull().And.HaveCount(2);
```

Reply:
Done